tests/http/test_07_upload.py

```diff
@@ -518,7 +518,7 @@ def test_07_41_post_urlenc_small(self, env: Env, httpd, nghttpx, proto):
         respdata = open(curl.response_file(0)).readlines()
         assert respdata == indata
 
-    def check_download(self, r, count, srcfile, curl):
+    def check_download(self, r: ExecResult, count: int, srcfile: str | os.PathLike, curl: CurlClient):
```

Comment:
IIRC, this | syntax is supported only in very recent Python versions, but, since I don't see a minimum Python version for curl listed in INTERNALS.md I suppose requiring a bleeding-edge Python doesn't go against any formal requirement. I think we really *should* add a Python version under *Build tools* in INTERNALS.md, though.

Reply:
I revert that back to a `Union[]` which works in older python3 versions. macOS seems to have a system python3 of 3.9.6 which is *almost* EOL according to [python.org/versions](https://devguide.python.org/versions/).